enterprise/server/raft/driver/driver.go

```diff
@@ -153,16 +170,244 @@ func computeQuorum(numNodes int) int {
 	return (numNodes / 2) + 1
 }
 
+type attemptRecord struct {
+	action          DriverAction
+	attempts        int
+	nextAttemptTime time.Time
+}
+
+type pqItem struct {
+	rangeID   uint64
+	replicaID uint64
+
+	priority   float64
+	insertTime time.Time
+
+	index      int // The index of the item in the heap.
+	processing bool
+	requeue    bool
+
+	attemptRecord attemptRecord
+}
+
+// An priorityQueue implements heap.Interface and holds pqItems.
+type priorityQueue []*pqItem
```

Comment:
nit: can you use the priority_queue we added in server/util/priority_queue instead of adding another one?